### PR TITLE
Log extension descriptor usage

### PR DESCRIPTION
**Pull request by boyan-velinov**
* Trace which extension descriptors are used
* Trace which extension descriptors are not used

Jira: LMCROSSITXSADEPLOY-1039

### CODE REVIEW

File: multiapps-controller-process/src/main/java/org/cloudfoundry/multiapps/controller/process/Messages.java [hunk 1]
```diff
@@ -674,8 +674,9 @@ public class Messages {
     public static final String PROCESS_WAS_DELETED_0 = "Process was deleted: {0}";
     public static final String MODULE_0_WAS_NOT_FOUND = "Module \"{0}\" was not found";
     public static final String DETECTING_LIVE_APPLICATION_ENV = "Detecting live application env...";
-    public static final String USED_EXTENSION_DESCRIPTORS_IDS = "Used extension descriptors with IDs: [{0}]";
-    public static final String NO_EXTENSION_DESCRIPTORS_PROVIDED = "No extension descriptors provided";
+    public static final String EXTENSION_DESCRIPTORS_IN_USE = "Extension descriptors with IDs in use: [{0}]";
```
Comment (IvanBorislavovDimitrov) [inline multiapps-controller-process/src/main/java/org/cloudfoundry/multiapps/controller/process/Messages.java:677]: Maybe:
Used extension descriptors for this deployment: ...
and:
Unused extension descriptors for this deployment: ...

File: multiapps-controller-process/src/main/java/org/cloudfoundry/multiapps/controller/process/steps/ProcessMtaExtensionDescriptorsStep.java [hunk 2]
```diff
@@ -75,26 +76,36 @@ private List<ExtensionDescriptor> parseExtensionDescriptors(String spaceId, List
...
-    private String createExtensionDescriptorsIdsMessage(List<ExtensionDescriptor> extensionDescriptorChain) {
-        return format(Messages.USED_EXTENSION_DESCRIPTORS_IDS, extensionDescriptorChain.stream()
-                                                                                       .map(ExtensionDescriptor::getId)
-                                                                                       .collect(Collectors.joining(", ")));
+        List<String> usedExtensionDescriptorIds = extensionDescriptorChain.stream()
+                                                                          .map(ExtensionDescriptor::getId)
+                                                                          .collect(Collectors.toList());
+
+        getStepLogger().info(format(Messages.EXTENSION_DESCRIPTORS_IN_USE,
+                                    String.join(", ", usedExtensionDescriptorIds)));
```
[the remark this answers is not in the document]
Comment (boyan-velinov) [reply multiapps-controller-process/src/main/java/org/cloudfoundry/multiapps/controller/process/steps/ProcessMtaExtensionDescriptorsStep.java]: done

File: multiapps-controller-process/src/main/java/org/cloudfoundry/multiapps/controller/process/steps/ProcessMtaExtensionDescriptorsStep.java [hunk 3]
```diff
@@ -75,26 +76,36 @@ private List<ExtensionDescriptor> parseExtensionDescriptors(String spaceId, List
...
+        
+        List<String> unusedExtensionDescriptors = providedExtensionDescriptors.stream()
+                                                                              .map(ExtensionDescriptor::getId)
+                                                                              .filter(id -> !usedExtensionDescriptorIds.contains(id))
+                                                                              .collect(Collectors.toList());
```
[the remark this answers is not in the document]
Comment (boyan-velinov) [reply multiapps-controller-process/src/main/java/org/cloudfoundry/multiapps/controller/process/steps/ProcessMtaExtensionDescriptorsStep.java:108]: ListUtil.select / selectRejected can be performed on elements of the same type, now we have List<ExtensionDescriptor> which will end up into List<String>, so working with streams seems more suitable.

File: multiapps-controller-process/src/main/java/org/cloudfoundry/multiapps/controller/process/Messages.java [hunk 4]
```diff
@@ -674,8 +674,9 @@ public class Messages {
...
+    public static final String EXTENSION_DESCRIPTORS_NOT_IN_USE = "Provided extension descriptors with IDs are not used: [{0}]";
```
[the remark this answers is not in the document]
Comment (boyan-velinov) [reply multiapps-controller-process/src/main/java/org/cloudfoundry/multiapps/controller/process/Messages.java]: Done - it was a mistake

**Review by IvanBorislavovDimitrov — APPROVED** [review]

**Pull request merged**